src/ScheduleFunctions.cpp

```diff
@@ -371,8 +373,11 @@ Stmt build_provide_loop_nest(const map<string, Function> &env,
                                      {StringImm::make(s.failure_message)},
                                      Internal::Call::Extern);
             then_case = AssertStmt::make(const_false(), specialize_fail_error);
+            // Since the condition is a constant true, the else case will
+            // never execute
+            else_case = then_case;
         }
-        stmt = IfThenElse::make(s.condition, then_case, stmt);
+        stmt = IfThenElse::make(s.condition, then_case, else_case);
```

Comment:
Isn't this constructing IfThenElse::make(s.condition, then_case, then_case)? Why not just set stmt = then_case in the else branch above and only construct the IfThenElse in the empty-failure-message branch.

Reply:
Very true.  Pushed a simpler version that doesn't construct a redundant IfThenElse.